BadApi/Models/Account/AccountModel.cs

```diff
@@ -0,0 +1,53 @@
+﻿using BadApi.Helpers;
+using BadApi.Models.Post;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Web;
+
+namespace BadApi.Models.Account
+{
+	public class AccountModel : Domain.Account
+	{
+		public IEnumerable<ViewPostModel> Posts
+		{
+			get
+			{
+				return Task.Run(() => AccountHelper.Instance.GetAccountPosts(AccountId, 0, 10)).GetAwaiter().GetResult();
+			}
+		}
+
+		public IEnumerable<ViewPostModel> Top5Posts
+		{
+			get
+			{
+				var allPosts = Task.Run(() => AccountHelper.Instance.GetAccountPosts(AccountId, 0, int.MaxValue))
```

Comment:
* Calling database from model - requires hack for calling async code synchronously and removes the ability to access per-thread state such as HttpContext in the methods being called.
* Fetching all records from the database
* Using int.MaxValue